src/com/baloise/sharedlib/tools/internal/MavenImpl.groovy

```diff
@@ -0,0 +1,27 @@
+package com.baloise.sharedlib.tools.internal
+
+import java.nio.file.*
+
+import com.baloise.sharedlib.common.Registered
+import com.baloise.sharedlib.tools.Maven
+
+class MavenImpl extends Registered implements Maven{
+
+	@Override
+	// tag see https://hub.docker.com/_/maven
+	public void mvn(String cmd, String tag = "") {
+		String image = tag ? "maven:$tag" : "maven"
+		if(steps.isUnix()) {
+			steps.docker.image(image).inside('-v $HOME/.m2:/root/.m2') {
+                steps.sh "mvn $cmd"
+			}
+		} else {
+			String sname = "docker-settings.xml"
+			String settings = 
+				Files.exists(Paths.get(System.getProperty("user.home"), '.m2').toRealPath().resolve(sname)) ? 
+				"-s /root/.m2/$sname" : 
+				""
+			steps.bat "docker run --rm -v \"%cd%\":/usr/src/mymaven -v \"%userprofile%\\.m2\":/root/.m2 -w /usr/src/mymaven $image mvn $settings $cmd"
```

Comment:
i vote against supporting windows. don't see the benefit when the whole build is run inside docker - why would i need a jenkins locally?

Reply:
as per decision 004 we do not support windows. I currently use this on my baloise box for development only. Made this explicit in https://github.com/baloise/CICD-pipeline/commit/cc54e42075d9f5ed62985a02ab621be3657396bb
See also https://github.com/baloise/CICD-pipeline/pull/25#discussion_r527834458 and #34